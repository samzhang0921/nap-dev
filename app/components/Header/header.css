.headerContainer {
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: space-around;
  border-top: 0;
  border-bottom: 1px dotted #999;
  float: none;
  clear: both;
  bottom: 8px;
  position: relative;
}
