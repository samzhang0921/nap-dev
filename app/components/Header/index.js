import React from 'react';
import style from './header.css';
import ListViewAndSort from './ListViewAndSort/ListViewAndSort';
import ListTitleAndTotal from './ListTitleAndTotal/ListTitleAndTotal';
import ListPagination from './ListPagination/ListPagination';
const Header = props => {

    return (
      <div className={style.headerContainer}>
        <ListViewAndSort totalPage={props.totalPage} currentPage={props.currentPage} changeProductsView = {props.changeProductsView} updateSort={props.updateSort}/>
        <ListTitleAndTotal total={props.total}/>
        <ListPagination totalPage={props.totalPage} currentPage={props.currentPage} updateOffset={props.updateOffset}/>
      </div>
    );
  }


export default Header;
