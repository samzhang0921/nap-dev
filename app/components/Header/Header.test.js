import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Header from './index';

describe('Header', () => {
  const offset = 60;
  const limit = 60;
  const total = 500;
  const totalPage =9;
  const currentPage = 1;
  const changeProductsView = jest.fn();
  const updateOffset = jest.fn();
  const updateSort = jest.fn();

  it('should mount item ', () => {
    const tree = toJson(mount(<Header />));
    expect(tree).toMatchSnapshot();
  });

  it('should match its snapshot with items', () => {

    const tree = renderer.create(
      <Header
    offset = {offset}
    limit = {limit}
    total = {total}
    totalPage ={totalPage}
    currentPage = {currentPage}
    changeProductsView = {changeProductsView}
    updateOffset = {updateOffset}
    updateSort = {updateSort}/>
  ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
