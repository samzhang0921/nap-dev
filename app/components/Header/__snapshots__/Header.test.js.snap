// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match its snapshot with items 1`] = `
<div
  className="headerContainer"
>
  <div
    className="ListViewAndSort"
  >
    <div>
      <button
        className="ListView"
        onClick={[Function]}
      >
        Product View
      </button>
    </div>
    <div
      className="sortBy"
    >
      <select
        name="sortOrder"
        onChange={[Function]}
      >
        <option
          value="sort"
        >
          Sort by
        </option>
        <option
          value="low"
        >
          Price Low to High
        </option>
        <option
          value="high"
        >
          Price High to Low
        </option>
      </select>
    </div>
  </div>
  <div
    className="ListTitleAndTotal"
  >
    <h1>
      what's new
    </h1>
    <div
      className="resultBox"
    >
      <div
        className="resultsBox"
      >
        <span>
          500
           Results
        </span>
      </div>
    </div>
  </div>
  <div
    className="pagenationBox"
  >
    <ul
      className="pagenationBoxUl"
    >
      <li
        className="pagination--1 pagenationBoxLi"
        onClick={[Function]}
      >
        <span>
          
        </span>
      </li>
      <li
        className="pagenationBoxCurrentPage"
        onClick={[Function]}
      >
        <span>
          1
        </span>
      </li>
      <li
        className="pagination-2 pagenationBoxLi"
        onClick={[Function]}
      >
        <span>
          2
        </span>
      </li>
      <li
        className="pagination-3 pagenationBoxLi"
        onClick={[Function]}
      >
        <span>
          3
        </span>
      </li>
      <li
        className="pagination-dot pagenationBoxLi"
        onClick={[Function]}
      >
        <span>
          ...
        </span>
      </li>
      <li
        className="pagination-9 pagenationBoxLi"
        onClick={[Function]}
      >
        <span>
          9
        </span>
      </li>
      <li
        className="pagination-0 pagenationBoxLi"
        onClick={[Function]}
      >
        <span>
          
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Header should mount item  1`] = `
<Header>
  <div
    className="headerContainer"
  >
    <ListViewAndSort>
      <div
        className="ListViewAndSort"
      >
        <ListView>
          <div>
            <button
              className="ListView"
              onClick={[Function]}
            >
              Product View
            </button>
          </div>
        </ListView>
        <ListSort>
          <div
            className="sortBy"
          >
            <select
              name="sortOrder"
              onChange={[Function]}
            >
              <option
                value="sort"
              >
                Sort by
              </option>
              <option
                value="low"
              >
                Price Low to High
              </option>
              <option
                value="high"
              >
                Price High to Low
              </option>
            </select>
          </div>
        </ListSort>
      </div>
    </ListViewAndSort>
    <ListTitleAndTotal>
      <div
        className="ListTitleAndTotal"
      >
        <h1>
          what's new
        </h1>
        <div
          className="resultBox"
        >
          <div
            className="resultsBox"
          >
            <span>
               Results
            </span>
          </div>
        </div>
      </div>
    </ListTitleAndTotal>
    <ListPagination>
      <div
        className="pagenationBox"
      >
        <ul
          className="pagenationBoxUl"
        >
          <PageNumber
            page={-1}
          >
            <li
              className="pagination--1 pagenationBoxLi"
              onClick={[Function]}
            >
              <span />
            </li>
          </PageNumber>
          <PageNumber
            page={1}
          >
            <li
              className="pagination-1 pagenationBoxLi"
              onClick={[Function]}
            >
              <span>
                1
              </span>
            </li>
          </PageNumber>
          <PageNumber
            page="..."
          >
            <li
              className="pagination-dot pagenationBoxLi"
              onClick={[Function]}
            >
              <span>
                ...
              </span>
            </li>
          </PageNumber>
          <PageNumber
            page={0}
          >
            <li
              className="pagination-0 pagenationBoxLi"
              onClick={[Function]}
            >
              <span />
            </li>
          </PageNumber>
        </ul>
      </div>
    </ListPagination>
  </div>
</Header>
`;
